Mono documentation

arschles opened the pull request:
TODO:
- [x] test upload and queue steps
- [ ] ~~add a config file, just as in the java examples~~

cc/ @treeder

arschles committed:
preliminary mono documentation

- upload and queue steps not tested
- no config file given yet

cc/ @treeder

treeder commented on mono/README.md:
````diff
@@ -0,0 +1,37 @@
+Now run it to test it out:
+
+```sh
+docker run --rm -v "$(pwd)":/worker -w /worker iron/mono:4.0.1 sh -c 'mono worker101.exe -payload hello.payload.json -id 123'
````
Change to env vars for payload.

treeder commented on mono/hello.cs:
```diff
@@ -0,0 +1,9 @@
+{
+    static public void Main ()
+    {
+        Console.WriteLine ("Hello Mono World");
```
Add code to use our .NET libs to read payload.

arschles replied:
@treeder where are our .Net libs? don't see anything on GH and this code taken from http://dev.iron.io/worker/languages/dotnet/#payload_example

arschles committed:
using env var for payload

I could have missed it, but I didn’t see support for grabbing a payload
in
https://github.com/iron-io/iron_dotnet/tree/master/src/IronSharp.IronWor
ker. this should do for now

cc/ @treeder

arschles committed:
removing unused example code

arschles commented:
@treeder this now uses an env var to get the payload file as requested. none of the client libraries listed in the dev center (http://grcodemonkey.github.io/iron_sharp/, https://github.com/odeits/IronTools and https://github.com/iron-io/iron_dotnet) seem to provide payload lookup so unable right now to follow [your request](https://github.com/iron-io/dockerworker/pull/21#discussion_r40608029). still combing through the code to make sure but it's a line of code to get the value of an env var in C# anyway.

please let me know if there are any other blockers to merging.

treeder commented:
Should update one of the dotnet libs (whichever is the best one) to read the env vars since that's our primary way of doing things now (regardless of language).

arschles commented:
@treeder it's on my todo list, but rather merge this now then update

treeder commented:
Ok, that doesn't look too bad like this anyways, couple lines, I like it.
